processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -163,7 +164,7 @@ public Object get()
       }
 
       @Override
-      public DimensionSelector makeDimensionSelector(final String dimension)
+      public DimensionSelector makeDimensionSelector(final String dimension, final ExtractionFn extractionFn)
```

Comment:
For topN and groupBy the actual lookups are performed only after doing the aggregation, and in those cases we already did the check for extraction function as part of the query engine before performing the lookup. Now it is performed within the selector instead of outside, so this should not affect performance for typical queries. 

The only queries that might be affected are aggregators that do the lookup as part of the object column selector to work on actual values, such as JavaScript and Cardinality. But in those instances a null  check is very cheap compared to the aggregation, and even then branch prediction will likely remove that overhead.